src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -701,12 +701,29 @@ private static int calculateTagColumn(Matcher javadocTagMatcher,
             final List<DetailAST> throwLiterals = findTokensInAstByType(blockAst,
                     TokenTypes.LITERAL_THROW);
             for (DetailAST throwAst : throwLiterals) {
-                final DetailAST newAst = throwAst.getFirstChild().getFirstChild();
-                if (newAst.getType() == TokenTypes.LITERAL_NEW) {
-                    final FullIdent ident = FullIdent.createFullIdent(newAst.getFirstChild());
-                    final ExceptionInfo exceptionInfo = new ExceptionInfo(
-                            createClassInfo(new Token(ident), currentClassName));
-                    returnValue.add(exceptionInfo);
+                DetailAST ancestor = throwAst;
+                while (ancestor != blockAst) {
+                    if (ancestor.getType() == TokenTypes.LITERAL_TRY
+                            || ancestor.getType() == TokenTypes.LAMBDA) {
+                        // if the throw is inside a try block or lambda, ignore it
+                        break;
+                    }
+                    if (ancestor.getType() == TokenTypes.LITERAL_CATCH
+                            || ancestor.getType() == TokenTypes.LITERAL_FINALLY) {
+                        // if the throw is inside a catch or finally block,
+                        // skip the immediate ancestor (try token)
+                        ancestor = ancestor.getParent();
+                    }
+                    ancestor = ancestor.getParent();
+                }
+                if (ancestor == blockAst) {
```

Comment:
It's better to extract this part to a method. And I would rename `blockAst` to `methodBodyAst` (or something similar).
And then:
```
if (isInTryOrLambda(methodBodyAst, throwAst)) {
...
}
```

Reply:
This section is now on its own method, with `methodBodyAst` as one of the parameter names.

But for the `getThrowed` method I feel it might be better to keep the existing name and avoid renaming it unnecessarily.